Moved building information into documentation

sntdevco opened the pull request:
The documentation for building and installing has been moved into a new Documentation section

azat commented on README.md:
```diff
@@ -84,95 +31,15 @@ Install CMake: <https://www.cmake.org>
-CMake also provides a GUI that lets you specify the source directory and output (binary) directory
-that the build should be placed in.
-
+See [Documentation/Building#Unix](/Documentation/Building.md#building-on-unix) for more information
```
This link is broken

sntdevco replied:
Rebased and fixed

ploxiln commented on README.md:
```diff
@@ -84,95 +31,15 @@ Install CMake: <https://www.cmake.org>
      $ start libevent.sln
 
+See [Documentation/Building#Windows](/Documentation/Building.md#building-on-windows) for more information
 ## CMake (Unix)
```
a blank line before headers would be good

sntdevco replied:
@ploxiln Review the most recent commit

ploxiln replied:
looks good to me, thanks

sntdevco commented:
@azat Will you be merging this in?

azat commented:
You mixed autoconf with cmake (`autogen.sh` note was under cmake build), so I rebased your patch a little (see d83b577d), objections?

sntdevco commented:
> You mixed autoconf with cmake (`autogen.sh` note was under cmake build), so I rebased your patch a little (see [d83b577](https://github.com/libevent/libevent/commit/d83b577d2bc1fc917474d9b10dd7b3220f0dd8f6)), objections?

**Objection :hand:**
I have rebased and pushed, you don't need to rebase individually

azat commented:
> Objection

What's the difference?

> I have rebased and pushed, you don't need to rebase individually

Information about flags for the autoconf still in the wrong section, plus I converted README to use ``` over `<tab>` for the markdown code blocks.

sntdevco commented:
@azat Just a point -> you could've mentioned this earlier while I rebased to fix the links, but anyways I've made whatever changes you wanted and now it's safe to be merged as I have rebased the patches to reflect the same

azat commented:
>Just a point -> you could've mentioned this earlier

I don't see any problems with me rebasing you patch (*you are still an author*).

And I gave you my patch/commit so that if you want to make this changes by your own, then you can take a look (*but looks like you didn't not look into it*), so *after 2 minutes of playing with links* here is how you can see it in github [web](https://github.com/libevent/libevent/compare/5e794493..d83b577d)

sntdevco commented:
> > Just a point -> you could've mentioned this earlier
> 
> I don't see any problems with me rebasing you patch (_you are still an author_).
> 
> And I gave you my patch/commit so that if you want to make this changes by your own, then you can take a look (_but looks like you didn't not look into it_), so _after 2 minutes of playing with links_ here is how you can see it in github [web](https://github.com/libevent/libevent/compare/5e794493..d83b577d)

Sorry I didn't understand what you meant! I'll take care the next time. Anyways, all good and rolling?

azat commented:
> I think this can be merged in now

If we are talking about d83b577d, then yep (since you had some disagreements, please confirm)

> all issues sorted?

Not all, but this is a good start

sntdevco commented:
> > I think this can be merged in now
> 
> If we are talking about [d83b577](https://github.com/libevent/libevent/commit/d83b577d2bc1fc917474d9b10dd7b3220f0dd8f6), then yep (since you had some disagreements, please confirm)
> 
> > all issues sorted?
> 
> Not all, but this is a good start

Go on

azat commented:
Applied, thanks!
